tenable/io/v3/was/plugins/api.py

```diff
@@ -38,9 +45,110 @@ def details(self, id: int) -> Dict:
         Examples:
             >>> plugin = tio.v3.was.plugins.details(1)
         '''
-        return self._get(f'{id}')
+        return self._details(f'{id}')
 
-    def search(self, **kwargs):
-        raise NotImplementedError(
-            'Search and Filter functionality will be updated later.'
-        )
+    def search(self,
+               **kw
+               ) -> Union[SearchIterator, CSVChunkIterator, Response]:
+        '''
+        Retrieves the plugins data
+        Args:
+            resource (str):
+                The json key to fetch the data from response
+            api_path (str):
+                API path for search endpoint
+            sort_type (enum):
+                Select format of sort expected by API. All the
+                supported formats are present in SortType Enumeration Class.
```

Comment:
I believe these should be removed from docstring since it would static for each endpoint